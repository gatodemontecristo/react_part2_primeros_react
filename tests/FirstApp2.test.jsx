const { render, screen } = require("@testing-library/react");
const { FirstApp } = require("../src/FirstApp");

describe('Pruebas en <FirstApp/>', () => { 

    const title = "Hola, soy Goku";
    const subTitle = 2023;
    test('debe de hacer match con el snapshot', () => { 
      const {container} = render(<FirstApp title={title} subTitle={subTitle}></FirstApp>);
      expect(container).toMatchSnapshot();
     });

     test('debe mostrar el mensaje "Hola, soy Goku', () => { 
       render(<FirstApp title={title} subTitle={subTitle}></FirstApp>);
      expect(screen.getByText(title)).toBeTruthy();
     });
 })