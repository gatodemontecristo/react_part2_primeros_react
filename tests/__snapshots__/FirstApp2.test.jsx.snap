// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    2023
  </p>
  <p>
    2023
  </p>
  <p>
    Fernando Herrera
  </p>
  <h1>
    La operacion mas basica es de 7
  </h1>
  <p>
    Spy un subtitulo
  </p>
</div>
`;
